### PR TITLE
Added code for Naive String Matching Algorithm

### PR DESCRIPTION
**Pull request by pritamnegi**
#247

### CODE REVIEW

File: Algorithms/String-Matching/NAIVE_STRING_SEARCHING.cpp [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+ 		}
+
+        if (j == pattern_length) 
+        {
+        	cout<< "Pattern string found at position "<< i<< endl;
+            flag = 1;
+        }
+    }
+
+    if(flag==0)
```
Comment (arnav-mandal1234) [inline Algorithms/String-Matching/NAIVE_STRING_SEARCHING.cpp]: space on both side of the operator

File: Algorithms/String-Matching/NAIVE_STRING_SEARCHING.cpp [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+	int pattern_length = pattern.length();
+
+	// Getting the length of text string
+    int text_length = text.length();
+ 	int flag = 0;
+    
+    // Loop to trace the complete text string
+    for (int i = 0; i <= text_length - pattern_length; i++)
+    {
+        int j;
```
Comment (arnav-mandal1234) [inline Algorithms/String-Matching/NAIVE_STRING_SEARCHING.cpp]: give i and j proper variable name.
It increases readability.

**Comment by pritamnegi** [comment]
Ohh sorry. I will rectify it.

File: Algorithms/String-Matching/NAIVE_STRING_SEARCHING.cpp (hunk 2 above)
Comment (pritamnegi) [reply Algorithms/String-Matching/NAIVE_STRING_SEARCHING.cpp]: What should I name them instead??

File: Algorithms/String-Matching/NAIVE_STRING_SEARCHING.cpp [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+{
+	// Getting the length of pattern string
+	int pattern_length = pattern.length();
+
+	// Getting the length of text string
+    int text_length = text.length();
+ 	int flag = 0;
+    
+    // Loop to trace the complete text string
+    for (int iterator1 = 0; iterator1 <= text_length - pattern_length; iterator1++)
```
Comment (arnav-mandal1234) [inline Algorithms/String-Matching/NAIVE_STRING_SEARCHING.cpp]: change it to text_iterator and another one as pattern_iterator.

File: Algorithms/String-Matching/NAIVE_STRING_SEARCHING.cpp [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+using namespace std;
+
+// Function to find the pattern string within the text string
+void NaiveStringMatching(string text, string pattern)
+{
+	// Getting the length of pattern string
+	int pattern_length = pattern.length();
+
+	// Getting the length of text string
+    int text_length = text.length();
```
Comment (arnav-mandal1234) [inline Algorithms/String-Matching/NAIVE_STRING_SEARCHING.cpp]: proper indentation.

File: Algorithms/String-Matching/NAIVE_STRING_SEARCHING.cpp [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+            if (text[text_iterator + pattern_iterator] != pattern[pattern_iterator])
+            {
+                break;
+            }
+ 		}
+
+        if (pattern_iterator == pattern_length) 
+        {
+        	cout<< "Pattern string found at position "<< text_iterator<< endl;
+            flag = 1;
```
Comment (arnav-mandal1234) [inline Algorithms/String-Matching/NAIVE_STRING_SEARCHING.cpp]: proper indentation.

**Comment by arnav-mandal1234** [comment]
squash the commits.

**Comment by pritamnegi** [comment]
I have squashed all the commits into one commit.

**Pull request merged**